### ISSUE TITLE
Network gas hourly and yearly calculations do not add up

### ISSUE THREAD

**Issue by kaskranenburgQ** [quintel/etsource#3080]
In a blank nl2019 scenario with the following input settings:
```
natural_gas_total_share: 0.0
green_gas_total_share: 100.0
propane_gas_total_share: 0.0
```
All extracted natural gas is now exported.

If I than compare the natural gas backup export hourly vs yearly with the following query:
```
EACH(
  V(energy_export_natural_gas_backup,demand),
  SUM(V(energy_export_natural_gas_backup,network_gas_input_curve))*3600,
  SUM(NEG(V(energy_export_natural_gas_backup,demand)),SUM(V(energy_export_natural_gas_backup,network_gas_input_curve))*3600)
  )
```
I get the following results:
![Screenshot 2024-06-28 at 17 05 12](https://github.com/quintel/etsource/assets/150331345/3bfbb137-bbc1-4c03-9bbe-da1222006b7f)
Indicating a delta of 4.8 PJ. 
This could occur due to a network gas node missing hourly calculation attributes

For the import of natural gas, we can see the following for a blank 2019 scenario:
 
<img width="892" alt="Screenshot 2024-07-10 at 09 02 28" src="https://github.com/quintel/etsource/assets/150331345/79a56726-3cd2-4b2d-b6b6-b9ff05c02764">

With this query :

`EACH(
  V(energy_import_natural_gas,demand),
 SUM(V(energy_import_natural_gas,network_gas_output_curve))*MJ_PER_MWH,
)
`
 A delta of 6.5 PJ

In addition, the energy_heat_backup_burner_ht_network_gas node shows a slight difference in energy flows:
Balancing issue in https://beta.energytransitionmodel.com/saved_scenarios/5126. 
Output of steam hot water:
```
EACH(
  V(energy_heat_flexibility_p2h_boiler_ht_electricity,output_of_steam_hot_water),
  SUM(V(energy_heat_flexibility_p2h_boiler_ht_electricity,steam_hot_water_output_curve))*MJ_PER_MWH,
  V(energy_heat_flexibility_p2h_heatpump_ht_electricity,output_of_steam_hot_water),
  SUM(V(energy_heat_flexibility_p2h_heatpump_ht_electricity,steam_hot_water_output_curve))*MJ_PER_MWH,
  V(energy_heat_backup_burner_ht_network_gas,output_of_steam_hot_water),
  SUM(V(energy_heat_backup_burner_ht_network_gas,steam_hot_water_output_curve))*MJ_PER_MWH
  )
```

Generates:

```
[
  33,278,159,838.49849,
  33,271,031,838.49849,
  16,639,079,919.249239,
  16,635,515,919.249243,
  295,714,984.85813123,
  295,714,984.8581313,
]
```
The node energy_heat_backup_burner_ht_network_gas has the following in the engine:
![[Screenshot 2025-01-09 at 13.59.23.png]]

The difference between 285.02 and 295.71 = 10.69
The difference on the energy_heat_flexibility_p2h_boiler_ht_electricity node is:  7.13
The difference on the energy_heat_flexibility_p2h_heatpump_ht_electricity node is : 3.56
Together this difference is equal to 10.69. 
Meaning that the p2h boiler and heatpump create the 'graph' inbalance on the energy_heat_backup_burner_ht_network_gas node. This is why the output we see that is shown in the graph is different from the actual node output.
This causes the sankey queries to become slightly inbalanced.

**Comment by mabijkerk** [comment]
Nice catch! Let's see if we can identify the culprit. If not, we may need to investigate the export and import type in reconciliation.

**Comment by kndehaan** [comment]
Find out whether demand is underspecified or supply is overspecified.

**Comment by mabijkerk** [comment]
@kaskranenburgQ isn't this actually the same larger issue as for https://github.com/quintel/etsource/issues/3048? Can we close on of both issues and rewrite the title to be more generci?

**Comment by mabijkerk** [comment]
According to one of our clients, the node `energy_export_natural_gas_backup` is missing from the hourly data-export of network gas. It may well be the case that this node is the cause of the mismatch in demand and supply. I suggest we start any further investigation of this issue at this node.

**Comment by kaskranenburgQ** [comment]
It might be that the reason that there is a mismatch lies in this issue #2144. 
In the cases we have drawn up we see the same as @michieldenhaan mentioned:

> The missing loss is either assigned to the import curve (in a scenario with import of natural gas) or to the export curve (in a scenario where domestic production is higher than demand).

**Comment by mabijkerk** [comment]
I'm also noticing that the `energy_export_greengas_backup` node has the type `consumer`, instead of `export`. This makes me wonder whether the hourly balancing works properly:

```
- use = undefined
- groups = [energy_export, application_group]
- free_co2_factor = 0.0
- network_gas.type = consumer
- network_gas.profile = network_gas_export
- network_gas.demand_carrier = greengas

~ demand = 0.0
```

**Comment by kaskranenburgQ** [comment]
> I'm also noticing that the energy_export_greengas_backup node has the type consumer, instead of export. 
If I recall correclty we did this since there can only be one network gas exporter (I can't remember why)
. This is the node `energy_export_natural_gas_backup`.

**Comment by mabijkerk** [comment]
Yes, but this is probably part of the reason why issues start to occur: if you only have green gas, how will the hourly calculations balance? I'll look into the graph design and hourly attributes to see if I can work out the logic.

**Comment by mabijkerk** [comment]
> I have a [scenario](https://energytransitionmodel.com/saved_scenarios/18678) where all heat production and demand is allocated to HT, however the network gas backup burner MT turns on without steam hot water output. Might this bug be related to all the above?

@kndehaan this is likely caused by the fact that DAC is connected in the graph to the HT network, but participates in the MT network hourly calculations.

**Comment by mabijkerk** [comment]
Overview of gas balancing issues:

- [ ] DAC: not showing up properly
- [ ] Export: missing from data export
- [ ] P2H: see https://github.com/quintel/etsource/issues/3165
- [ ] Treatment: see https://github.com/quintel/etsource/issues/2144
- [ ] Backup burner: t.d.b.

**Comment by mabijkerk** [comment]
It seems a large part of the imbalance between annual and hourly values is caused by the gas power fuelmix. In a local `nl2019` scenario with losses from natural gas treatment set to zero, the annual and hourly import is queried:

```
EACH(
  V(energy_import_natural_gas,output_of_natural_gas),
  SUM(V(energy_import_natural_gas,network_gas_output_curve))*MJ_PER_MWH
)
```
First with a completely blank scenario:
```
[
  493,239,088,457.2776,
  520,899,250,749.96313,
]
```
Then a scenario with the share of network gas in the gas power fuelmix set to 100%:
```
[
  520,372,089,527.83545,
  520,372,089,527.84467,
]
```

It is clear that the gas power fuelmix (in addition to the treatment losses) is a main culprit in the difference between annual and hourly balancing of network gas. Further investigation is needed to determine why gas power fuelmix is not handled properly.

**Comment by mabijkerk** [comment]
All open issue with network gas balancing have been addressed in other issues. This general issue is therefore closed.
